templates/broker.xml.erb

```diff
@@ -0,0 +1,96 @@
+<?xml version='1.0' encoding="UTF-8"?>
+<!--
+  File managed with puppet ###
+  Module: '<%= scope.to_hash['module_name'] %>'
+-->
+<configuration xmlns="urn:activemq"
+               xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+               xsi:schemaLocation="urn:activemq /schema/artemis-configuration.xsd">
+
+    <core xmlns="urn:activemq:core" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:schemaLocation="urn:activemq:core ">
+
+
+        <acceptors>
+            <acceptor name="in-vm">vm://0</acceptor>
+            <acceptor name="stomp">tcp://<%= @artemis_host %>:<%= @artemis_port %>?protocols=STOMP;sslEnabled=true;trustStorePath=<%= @truststore_file %>;trustStorePassword=<%= @truststore_password %>;keyStorePath=<%= @keystore_file %>;keyStorePassword=<%= @keystore_password %>;needClientAuth=true</acceptor>
+        </acceptors>
+
+
+        <security-enabled>false</security-enabled>
+
+        <!-- Silence warnings on server startup -->
+        <cluster-user></cluster-user>
+        <cluster-password></cluster-password>
+
+        <persistence-enabled>true</persistence-enabled>
+        <journal-type>NIO</journal-type>
+        <journal-pool-files>1</journal-pool-files>
+        <create-bindings-dir>true</create-bindings-dir>
+        <create-journal-dir>true</create-journal-dir>
+        <bindings-directory>/var/lib/candlepin/activemq-artemis/bindings</bindings-directory>
+        <journal-directory>/var/lib/candlepin/activemq-artemis/journal</journal-directory>
+        <large-messages-directory>/var/lib/candlepin/activemq-artemis/largemsgs</large-messages-directory>
+        <paging-directory>/var/lib/candlepin/activemq-artemis/paging</paging-directory>
+
+        <!-- Use default thread pool max sizes as defined by Artemis. -->
+        <!-- Uncomment to customize.                                  -->
+        <!-- <thread-pool-max-sie>30</thread-pool-max-size> -->
+        <!-- <scheduled-thread-pool-max-size>5</scheduled-thread-pool-max-size> -->
+
+        <addresses>
+            <address name="event.default">
+                <multicast>
+                    <queue name="event.org.candlepin.audit.LoggingListener" />
+                    <queue name="event.org.candlepin.audit.ActivationListener" />
+                </multicast>
+            </address>
+
+            <address name="katello.candlepin">
```

Comment:
Since the configuration of this address needs to be in agreement with Katello's configuration for the event listener, I wonder if they should originate in puppet-katello and this address block can be added dynamically? Similar for the 'divert' below.

Reply:
If I understand correctly, Artemis is part of candlepin. If we want users to use this module to manage candlepin outside if the katello scope, it would be good if this was configurable.
Basically, this is an erb template. So you can do whatever magic you want. Defining the `addresses` in a hash in `puppet-katello` and passing this to this module is quite easy.

Reply:
I don't honestly expect anyone to use this module to deploy Candlepin outside the scope of Katello. I can make this configurable, I just tend to think for its primary use case, all of that abstraction can simply make this harder to understand and make changes too.